fuse/ipns/ipns_unix.go

```diff
@@ -168,8 +154,27 @@ func (s *Root) Lookup(ctx context.Context, name string) (fs.Node, error) {
 	return &Link{s.IpfsRoot + "/" + resolved.B58String()}, nil
 }
 
-// ReadDirAll reads a particular directory. Disallowed for root.
+func (r *Root) Close() error {
+	for _, kr := range r.Roots {
+		err := kr.Publish(r.Ipfs.Context())
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Forget is called when the filesystem is unmounted. probably.
```

Comment:
Yeah, i might actually remove this method, the fuse documentation is very unclear about when this is called and what it signifies.